pkg/api/image/formatter.go

```diff
@@ -60,18 +62,60 @@ func (h UploadActionHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h UploadActionHandler) do(rw http.ResponseWriter, req *http.Request) error {
+func (h ImageHandler) do(rw http.ResponseWriter, req *http.Request) error {
 	vars := mux.Vars(req)
-	action := vars["action"]
-	switch action {
-	case actionUpload:
+	operation := ""
+	if _, found := vars["action"]; found {
+		operation = vars["action"]
+	} else if _, found := vars["link"]; found {
+		operation = vars["link"]
+	}
+	switch operation {
+	case operationUpload:
```

Comment:
wondering, if mixing `action` and `operation` is better

vars["action"] represents `upload`, vars["link"] represent `download`

when keeping name `action` unchanged, the only tricky is that when vars["links"] is existing, the `action` is changed from `upoad` to `download`, but it's still an action


when reading L40 `resource.AddAction(request, operationUpload)` , it will be a bit confusing

Reply:
Already rewrite here.

I prefer to use `operation` because maybe we need to implement another image operation in the feature.
The `action` means the operation is`POST` method.
The `link` means the operation is`GET` method.

That, the `operation` to present what we want to do next. No matter where it comes from `action` or `link`.

Reply:
Please take a look of other related files, all are following `action` naming.  It's better for `image/formatter` also in same line, even though it may not fully meet your expectation.
 
That will not confuse other engineer when they read&mantain code.

```
node/formatter.go
volume/handler.go
vm/handler.go
...
```

Reply:
Already change back to `action`. Please check it again, thanks :)